casper/src/main/rholang/MakePoS.rho

```diff
@@ -61,6 +62,57 @@ contract @"MakePoS"(@purse, @minimumBond, @maximumBond, @initBonds, return) = {
       }
     } |
     
+    contract feeDistribution(@n, @joiningFee, return) = {
+      new rangeCh, feeDistFunc, feeDistCh, sum, nCh, totalCh in {
+        contract sum(@x, @y, return) = { return!(x + y) } |
```

Comment:
I don't know if I totally agree. It might be useful to have some common combinators in the registry, but I think it would be even more useful to be able to create "anonymous contracts" easily in-line. Basically I just really want that functional sub-language @pyrocto keeps mentioning.

Reply:
I like the in-line contracts :+1:

Reply:
On the other hand, we usually want to signal the result or end of the computation to sequence them so it's harder.